pallets/automation-time/src/lib.rs

```diff
@@ -764,33 +772,58 @@ pub mod pallet {
 			<T as Config>::WeightInfo::run_native_transfer_task()
 		}
 
+		fn is_completed_task(time: BoundedVec<UnixTime, T::MaxRecurringTimes>) -> bool {
+			let current_time_slot = match Self::get_current_time_slot() {
+				Ok(time_slot) => time_slot,
+				Err(_) => return false, // fix error
+			};
+
+			let mut scheduled_times = time.to_vec();
+			scheduled_times.sort_unstable();
+			scheduled_times.dedup();
+
+			return *scheduled_times.last().unwrap() <= current_time_slot;
+		}
+
 		fn remove_task(task_id: T::Hash, task: Task<T>) {
+			let current_time_slot = match Self::get_current_time_slot() {
+				Ok(time_slot) => time_slot,
+				Err(_) => return, // fix error
+			};
 			let mut found_task: bool = false;
-			match Self::get_scheduled_tasks(task.time) {
-				None => {
-					let mut task_queue = Self::get_task_queue();
-					for i in 0..task_queue.len() {
-						if task_queue[i] == task_id {
-							task_queue.remove(i);
-							TaskQueue::<T>::put(task_queue);
-							found_task = true;
-							break
-						}
-					}
-				},
-				Some(mut task_ids) =>
-					for i in 0..task_ids.len() {
-						if task_ids[i] == task_id {
-							if task_ids.len() == 1 {
-								<ScheduledTasks<T>>::remove(task.time);
-							} else {
-								task_ids.remove(i);
-								<ScheduledTasks<T>>::insert(task.time, task_ids);
+			let mut times: Vec<UnixTime> = task.time.to_vec();
+
+			times.sort_unstable();
+			times.dedup();
+
+			for scheduled_time in times.iter().rev() {
+				if *scheduled_time < current_time_slot { break; }
+				match Self::get_scheduled_tasks(*scheduled_time) {
```

Comment:
not sure i understand this match statement. i get that it removes from scheduled tasks if scheduled time is found there. but why only remove from task queue if scheduled time is not found?

Reply:
this made sense when you could only schedule for one time, so either your task is in `ScheduledTasks` or `TaskQueue`.  I'll need to update this function because now you could have tasks in both places at once.